File: 1791-find-center-of-star-graph/1791-find-center-of-star-graph.js
/**
 * @param {number[][]} edges
 * @return {number}
 */
var findCenter = function (edges) {
    let outdegree = new Map();
    let ans;

    for (let [from, to] of edges) {
        outdegree.set(from, (outdegree.get(from) ?? 0) + 1);
        outdegree.set(to, (outdegree.get(to) ?? 0) + 1);
    }

    for (let [key, value] of outdegree) {
        if (value === outdegree.size - 1) {
            ans = key;
            break;
        }
    }

    return ans;
};

File: 1791-find-center-of-star-graph/1791-find-center-of-star-graph.py
class Solution:
    def findCenter(self, edges: List[List[int]]) -> int:
        outdegree = {}

        for frm, to in edges:
            outdegree[frm] = outdegree.get(frm, 0) + 1
            outdegree[to] = outdegree.get(to, 0) + 1

        for key, val in outdegree.items():
            if val == len(outdegree) - 1:
                return key